src/test/java/core/basesyntax/service/process/impl/FruitTransactionServiceImplTest.java

```diff
@@ -0,0 +1,135 @@
+package core.basesyntax.service.process.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.impl.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operations.BalanceHandler;
+import core.basesyntax.service.operations.OperationHandler;
+import core.basesyntax.service.operations.PurchaseHandler;
+import core.basesyntax.service.operations.ReturnHandler;
+import core.basesyntax.service.operations.SupplyHandler;
+import core.basesyntax.service.process.FruitTransactionService;
+import core.basesyntax.service.strategy.OperationStrategy;
+import core.basesyntax.service.strategy.impl.OperationStrategyImpl;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitTransactionServiceImplTest {
+    private static final StorageDao storageDao = new StorageDaoImpl();
+    private static Map<FruitTransaction.Operation,
+            OperationHandler> operationOperationsHandlerMap = new HashMap<>();
+    private static OperationStrategy operationStrategy =
+            new OperationStrategyImpl(operationOperationsHandlerMap);
+    private static final FruitTransactionService fruitTransactionService =
+            new FruitTransactionServiceImpl(operationStrategy);
+
+    @BeforeClass
+    public static void beforeClass() {
+        operationOperationsHandlerMap.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseHandler(storageDao));
+        operationOperationsHandlerMap.put(FruitTransaction.Operation.RETURN,
+                new ReturnHandler(storageDao));
+        operationOperationsHandlerMap.put(FruitTransaction.Operation.BALANCE,
+                new BalanceHandler(storageDao));
+        operationOperationsHandlerMap.put(FruitTransaction.Operation.SUPPLY,
+                new SupplyHandler(storageDao));
+    }
+
+    @After
+    public void tearDown() {
+        Storage.storage.clear();
+    }
+
+    @Test
+    public void processTransactions_balance_ok() {
+        FruitTransaction.Operation balanceOperation = FruitTransaction.Operation.BALANCE;
+        String fruitName = "banana";
```

Comment:
You can make this file constant, cause you use this in most of your tests. Check other